Source/Task/AsyncLib.cpp

```diff
@@ -548,35 +640,21 @@ STDAPI GetAsyncResult(
         else if (token != state->token)
         {
             // Call/Result mismatch.  This AsyncBlock was initiated by state->function
-            char buf[100];
-            if (state->function != nullptr)
-            {
-                sprintf_s(
-                    buf,
-                    "Call/Result mismatch.  This AsyncBlock was initiated by '%s'.\r\n",
-                    state->function);
-            }
-            else
-            {
-                sprintf_s(buf, "Call/Result mismatch\r\n");
-            }
-
-            result = E_INVALIDARG;
-            ASYNC_LIB_TRACE(result, buf);
             ASSERT(false);
+            result = E_INVALIDARG;
         }
         else if (state->providerData.bufferSize == 0)
         {
             // Caller has not supplied a payload
-            result = HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED);
+            result = E_HC_FEATURE_NOT_PRESENT;
         }
         else if (buffer == nullptr)
         {
             return E_INVALIDARG;
         }
         else if (bufferSize < state->providerData.bufferSize)
         {
-            return HRESULT_FROM_WIN32(ERROR_INSUFFICIENT_BUFFER);
+            return E_HC_BUFFER_TOO_SMALL;
```

Comment:
Are these defined to be the same values as ERROR_NOT_SUPPORT and ERROR_INSUFFFICIENT_BUFFER?  I'll need to define constants for these on the Windows side and I don't want to leak new values if we already have error codes that work.